plugin/yt_dlp_plugins/extractor/getpot_bgutil_script.py

```diff
@@ -51,8 +56,11 @@ def _validate_get_pot(self, client: str, ydl: YoutubeDL, visitor_data=None, data
     def _get_pot(self, client: str, ydl: YoutubeDL, visitor_data=None, data_sync_id=None, player_url=None, **kwargs) -> str:
         self._logger.info(
             f'Generating POT via script: {self.script_path}')
+        self._logger.debug(f'Proxies: {self.proxies!r}')
 
         command_args = ['node', self.script_path]
+        if proxy := self.proxies:  # maybe?
+            command_args.extend(['-p', ','.join(proxy.values())])
```

Comment:
maybe we should just use proxy.get('all') since we only declared support for 'all' proxy

Reply:
I'm a bit confused by what `Features.ALL_PROXY` actually does, because the other enum is `Features.NO_PROXY`. It also looks like you can set a proxy for a specific URL protocol as well instead of 'all' (https://github.com/yt-dlp/yt-dlp/blob/4a9bc8c3630378bc29f0266126b503f6190c0430/test/test_networking.py#L1575). We should still allow for 'http' and 'https'. Not sure if `proxy.get('all')` would do that

Reply:
> I'm a bit confused by what `Features.ALL_PROXY` actually does, because the other enum is `Features.NO_PROXY`.

from my understanding, all proxy stands for global proxy/use proxy to handle all traffic, while no proxy means the RH can choose not to use proxy when there is one(?)

is that proxies in `http`/`http` key from env vars(`HTTP_PROXY`, `HTTPS_PROXY`)?
seems like it always goes to `all` when I pass the proxy with `--proxy`

Reply:
![Screenshot_20240918_141456](https://github.com/user-attachments/assets/76996748-3bd6-494e-b853-6b4002a46f36)

Reply:
So in that case, `proxies.get('all')` wouldn't achieve what we want? Need to figure out how we want to prioritize them

Reply:
marking this as resolved